client/lib/login/index.js

```diff
@@ -55,33 +55,21 @@ export function pathWithLeadingSlash( path ) {
 
 export function getSignupUrl( currentQuery, currentRoute, oauth2Client, locale, pathname ) {
 	const signupUrl = config( 'signup_url' );
-
 	const redirectTo = get( currentQuery, 'redirect_to', '' );
-	const signupFlow = get( currentQuery, 'signup_flow' );
-	const wccomFrom = get( currentQuery, 'wccom-from' );
-	const isFromMigrationPlugin = includes( redirectTo, 'wpcom-migration' );
-
-	/**
-	 *  Include redirects to public.api/connect/?action=verify&service={some service}
-	 *  If the signup is from the Highlander Comments flow, the signup page will be in a popup modal
-	 *  We need to redirect back to public.api/connect/ to do an external login and close modal
-	 *  Ref: PCYsg-Hfw-p2
-	 */
-	const isFromPublicAPIConnectFlow = includes( redirectTo, 'public.api/connect/?action=verify' );
 
 	if (
 		// Match locales like `/log-in/jetpack/es`
 		startsWith( currentRoute, '/log-in/jetpack' )
 	) {
 		// Basic validation that we're in a valid Jetpack Authorization flow
 		if (
-			includes( get( currentQuery, 'redirect_to' ), '/jetpack/connect/authorize' ) &&
-			includes( get( currentQuery, 'redirect_to' ), '_wp_nonce' )
+			includes( redirectTo, '/jetpack/connect/authorize' ) &&
+			includes( redirectTo, '_wp_nonce' )
```

Comment:
Here and at a few other places I'm reusing the existing `redirectTo` value. It's guaranteed to be non-empty because it satisfied checks for a substring.